recipes/libxml2/build.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [[ `uname` == 'Darwin' ]];
+then
+    export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
+elif [[ `uname` == 'Linux' ]];
+then
+    export LIBRARY_SEARCH_VAR=LD_LIBRARY_PATH
+fi
+
+./autogen.sh
+./configure --prefix="${PREFIX}" \
+            --with-iconv="${PREFIX}" \
+            --with-icu \
+            --with-lzma="${PREFIX}" \
+            --with-python="${PREFIX}" \
+            --with-zlib="${PREFIX}"
+make
+# Correct weirdly linked library paths.
+if [[ `uname` == 'Darwin' ]];
+then
+    LIBDIR="${PREFIX}/lib"
+    LIBICUDATA="`cd ${LIBDIR} && ls libicudata.*.*.dylib`"
+    find "${SRC_DIR}/.libs" -type f | xargs -I {} install_name_tool -change "../lib/${LIBICUDATA}" "${PREFIX}/lib/${LIBICUDATA}" {} || true
+    find "${SRC_DIR}/python/.libs" -type f | xargs -I {} install_name_tool -change "../lib/${LIBICUDATA}" "${PREFIX}/lib/${LIBICUDATA}" {} || true
```

Comment:
Here is where the magic happens. Basically we take each file in `.libs` (includes all executables and binaries that are built) and `python/.libs` (includes all libraries built for use with Python like modules) that we find and we change any name that we find that contains `../lib/libicudata.*.*.dylib` to the full absolute path to `libicudata.*.*.dylib`. Some files we find are not linked or cannot be linked and so we silence any exit code we would get from these as they are just no-ops. As a result, the libraries are now referenced as they should be by absolute paths. This is how every other library would be normally referenced at this point. Now everything proceeds normally.